AdminUi/apps/admin_ui/lib/home/identity_details/identity_details.dart

```diff
@@ -216,3 +219,225 @@ class _IdentityDetails extends StatelessWidget {
+                                    quota.period,
+                                    style: TextStyle(color: shouldDisable ? Colors.grey : null),
+                                  ),
+                                ),
+                                DataCell(
+                                  Tooltip(
+                                    message: tooltipMessage ?? '',
+                                    child: isTierQuota
+                                        ? Icon(
+                                            Icons.info,
+                                            color: shouldDisable ? Colors.grey : null,
+                                          )
+                                        : null,
+                                  ),
+                                ),
+                              ],
+                            );
+                          },
+                        ),
+                      ];
+                    }).toList(),
+                  ),
+                ),
+              ],
+            ),
+          ),
+        ],
+      ),
+    );
+  }
+
+  Map<String, List<IdentityQuota>> _groupQuotas() {
+    final groupedQuotas = <String, List<IdentityQuota>>{};
+
+    if (widget.identityDetails.quotas != null) {
+      for (final quota in widget.identityDetails.quotas!) {
+        if (groupedQuotas.containsKey(quota.metric.displayName)) {
+          groupedQuotas[quota.metric.displayName]!.add(quota);
+        } else {
+          groupedQuotas[quota.metric.displayName] = [quota];
+        }
+      }
+    }
+
+    return groupedQuotas;
+  }
+
+  void _toggleSelection(String id) {
+    setState(() {
+      if (_selectedQuotas.contains(id)) {
+        _selectedQuotas.remove(id);
+        return;
+      }
+
+      _selectedQuotas.add(id);
+    });
+  }
+
+  Future<void> _removeSelectedQuotas() async {
+    final confirmed = await showConfirmationDialog(
+      context: context,
+      title: 'Remove Quotas',
+      message: 'Are you sure you want to remove the selected quotas from the identity "${widget.identityDetails.address}"?',
+    );
+
+    if (!confirmed) return;
+
+    for (final quota in _selectedQuotas) {
+      final result =
+          await GetIt.I.get<AdminApiClient>().quotas.deleteIdentityQuota(address: widget.identityDetails.address, individualQuotaId: quota);
+      if (result.hasError && mounted) {
+        ScaffoldMessenger.of(context).showSnackBar(
+          const SnackBar(
+            content: Text('An error occurred while deleting the quota(s). Please try again.'),
+            showCloseIcon: true,
+          ),
+        );
+        return;
+      }
+
+      widget.onQuotasChanged();
+    }
+
+    _selectedQuotas.clear();
+    if (mounted) {
+      ScaffoldMessenger.of(context).showSnackBar(
+        const SnackBar(
+          content: Text('Selected quota(s) have been removed.'),
+          showCloseIcon: true,
+        ),
+      );
+    }
+  }
+}
```

Comment:
please move this to a separate file, maybe you can also extract a bit of code out of the widgets build method as this is also kind of big ..